src/platform/packages/shared/kbn-workflows/spec/lib/get_json_schema_from_yaml_schema.ts

```diff
@@ -0,0 +1,193 @@
+    '"type": "object", "properties": {}, "additionalProperties": false, "description": "Complex nested configuration (simplified due to schema complexity)"'
+  );
+
+  // Pattern 3: Fix 3-level allOf references with array items
+  // Examples: allOf/1/allOf/0/properties/threat_mapping/items, allOf/1/allOf/0/properties/new_terms_fields/items
+  fixedSchemaString = fixedSchemaString.replace(
+    /"\$ref":"#\/definitions\/WorkflowSchema\/properties\/settings\/properties\/on-failure\/properties\/fallback\/items\/anyOf\/\d+\/properties\/with\/anyOf\/\d+\/allOf\/\d+\/allOf\/\d+\/properties\/[^"]+\/items"/g,
+    '"type": "array", "items": {"type": "object", "additionalProperties": false}, "description": "Array configuration (simplified)"'
+  );
+
+  // Pattern 4: Fix 3-level allOf references for regular properties
+  fixedSchemaString = fixedSchemaString.replace(
+    /"\$ref":"#\/definitions\/WorkflowSchema\/properties\/settings\/properties\/on-failure\/properties\/fallback\/items\/anyOf\/\d+\/properties\/with\/anyOf\/\d+\/allOf\/\d+\/allOf\/\d+\/properties\/[^"]+"/g,
+    '"type": "object", "properties": {}, "additionalProperties": false, "description": "Nested configuration (simplified)"'
+  );
+
+  // Pattern 5: Fix any remaining deeply nested broken references with properties
+  fixedSchemaString = fixedSchemaString.replace(
+    /"\$ref":"#\/definitions\/[^"]*\/allOf\/\d+\/allOf\/\d+\/allOf\/\d+\/properties\/[^"]+"/g,
+    '"type": "object", "properties": {}, "additionalProperties": false, "description": "Complex object (validation simplified)"'
+  );
+
+  // Pattern 6: Fix any remaining multi-level allOf references with properties
+  fixedSchemaString = fixedSchemaString.replace(
+    /"\$ref":"#\/definitions\/[^"]*\/allOf\/\d+\/allOf\/\d+\/properties\/[^"]+"/g,
+    '"type": "object", "properties": {}, "additionalProperties": false, "description": "Nested object (validation simplified)"'
+  );
+
+  // Pattern 7: Fix any remaining bare allOf references (catch-all)
+  fixedSchemaString = fixedSchemaString.replace(
+    /"\$ref":"#\/definitions\/[^"]*\/allOf\/\d+\/allOf\/\d+(?:\/allOf\/\d+)*"/g,
+    '"type": "object", "properties": {}, "additionalProperties": false, "description": "Schema intersection (simplified due to broken reference)"'
+  );
+
+  // Enforce strict validation: ensure all objects have additionalProperties: false
+  // This fixes the main issue where Kibana connectors were too permissive
+  try {
+    const tempSchema = JSON.parse(fixedSchemaString);
+    fixAdditionalPropertiesInSchema(tempSchema);
+    fixedSchemaString = JSON.stringify(tempSchema);
+  } catch (parseError) {
+    // If parsing fails, throw an error, since replacing with regexp isn't safe
+    throw new Error('Failed to fix additionalProperties in json schema');
+  }
+
+  try {
+    const fixedSchema = JSON.parse(fixedSchemaString);
```

Comment:
Can we avoid this parse? It looks like we have already parsed the string right above

Reply:
Well, you are also right that we are not parsing what comes out of the `fixAdditionalPropertiesInSchema`, since we are applying a bunch of regex, maybe the parse is needed...

I am not sure, this logic is definitely a bit funky and not the easiest to reason through. But I guess it's necessary, and since it only runs once when connectors are loaded, I'll approve